CondFormats/GEMObjects/src/GEMEMap.cc

```diff
@@ -0,0 +1,101 @@
+#include "CondFormats/GEMObjects/interface/GEMEMap.h"
+#include "CondFormats/GEMObjects/interface/GEMROmap.h"
+#include "DataFormats/MuonDetId/interface/GEMDetId.h"
+
+GEMEMap::GEMEMap():
+  theVersion("") {}
+
+GEMEMap::GEMEMap(const std::string & version):
+  theVersion(version) {}
+
+GEMEMap::~GEMEMap() {}
+
+const std::string & GEMEMap::version() const{
+  return theVersion;
+}
+
+GEMROmap* GEMEMap::convert() const{
+  GEMROmap* romap=new GEMROmap();
+
+  for (auto imap : theVFatMaptype){
+    for (unsigned int ix=0;ix<imap.strip_number.size();ix++){
+      GEMROmap::eCoord ec;
+      ec.vfatId= imap.vfatId[ix] - 0xf000;// chip ID is 12 bits
+      ec.channelId=imap.vfat_chnnel_number[ix];
+      GEMROmap::dCoord dc;
+      dc.stripId = 1 + imap.strip_number[ix]+(imap.iPhi[ix]-1)%3*128;
+      dc.gemDetId = GEMDetId(imap.z_direction[ix], 1, std::abs(imap.z_direction[ix]), imap.depth[ix], imap.sec[ix], imap.iEta[ix]); 
+      romap->add(ec,dc);
+      romap->add(dc,ec);
+    }
+  }
+  return romap;
+}
+
+GEMROmap* GEMEMap::convertDummy() const{
+  GEMROmap* romap=new GEMROmap();
+  uint16_t amcId = 1; //amc
+  uint16_t gebId = 1; 
+  romap->addAMC(amcId);
+	
+  for (int re = -1; re <= 1; re = re+2) {
+    for (int st = GEMDetId::minStationId; st<=GEMDetId::maxStationId; ++st) {
+      for (int ch = 1; ch<=GEMDetId::maxChamberId; ++ch) {
+      	for (int ly = 1; ly<=GEMDetId::maxLayerId; ++ly) {
+	  
+	  // 1 geb per chamber
+	  // 24 gebs per amc
+	  // make new amcId once 24 gebs are used up
+	  if (gebId > 25){
+	    gebId = 1;
+	    amcId++;
+	    romap->addAMC(amcId);
+	  }
+
+	  romap->addAMC2GEB(amcId, gebId);
+	  
+	  GEMDetId chamDetId(re, 1, st, ly, ch, 0);
+	  uint32_t chamberId = (amcId << 5) | gebId;	  
+	  romap->add(chamDetId,chamberId);
+	  romap->add(chamberId,chamDetId);
+	  
+	  uint16_t chipId = 0;	 	  
+	  for (int roll = 1; roll<=GEMDetId::maxRollId; ++roll) {
+	    int maxVFat = 3;
```

Comment:
magic number?
(also, if this is only based on the station, it could be defined in the outer loop)